src/m365/entra/commands/pim/pim-role-assignment-add.ts

```diff
@@ -0,0 +1,319 @@
+
+        if (args.options.endDateTime && !validation.isValidISODateTime(args.options.endDateTime)) {
+          return `${args.options.endDateTime} is not a valid ISO 8601 date time string`;
+        }
+
+        if (args.options.duration && !validation.isValidISODuration(args.options.duration)) {
+          return `${args.options.duration} is not a valid ISO 8601 duration`;
+        }
+
+        if (args.options.administrativeUnitId && !validation.isValidGuid(args.options.administrativeUnitId)) {
+          return `${args.options.administrativeUnitId} is not a valid GUID`;
+        }
+
+        if (args.options.applicationId && !validation.isValidGuid(args.options.applicationId)) {
+          return `${args.options.applicationId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['roleDefinitionName', 'roleDefinitionId'] });
+    this.optionSets.push({
+      options: ['noExpiration', 'endDateTime', 'duration'],
+      runsWhen: (args) => {
+        return !!args.options.noExpiration || args.options.endDateTime !== undefined || args.options.duration !== undefined;
+      }
+    });
+    this.optionSets.push({
+      options: ['userId', 'userName', 'groupId', 'groupName'],
+      runsWhen: (args) => {
+        return args.options.userId !== undefined || args.options.userName !== undefined || args.options.groupId !== undefined || args.options.groupName !== undefined;
+      }
+    });
+    this.optionSets.push({
+      options: ['administrativeUnitId', 'applicationId'],
+      runsWhen: (args) => {
+        return args.options.administrativeUnitId !== undefined || args.options.applicationId !== undefined;
+      }
+    });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    let action = 'adminAssign';
+
+    try {
+      const token = auth.connection.accessTokens[auth.defaultResource].accessToken;
+      const isAppOnlyAccessToken = accessToken.isAppOnlyAccessToken(token);
+      if (isAppOnlyAccessToken) {
+        throw 'When running with application permissions either userId, userName, groupId or groupName is required';
+      }
+
+      const roleDefinitionId = await this.getRoleDefinitionId(args.options, logger);
+      const principalId = await this.getPrincipalId(args.options, logger);
+
+      if (!args.options.userId && !args.options.userName && !args.options.groupId && !args.options.groupName) {
+        action = 'selfActivate';
+      }
+
+      const requestOptions: CliRequestOptions = {
+        url: `${this.resource}/v1.0/roleManagement/directory/roleAssignmentScheduleRequests`,
+        headers: {
+          'accept': 'application/json;odata.metadata=none'
+        },
+        responseType: 'json',
+        data: {
+          principalId: principalId,
+          roleDefinitionId: roleDefinitionId,
+          directoryScopeId: this.getDirectoryScope(args.options),
+          action: action,
```

Comment:
We don't need the separate action variable I think. We can just add the condition here:

```suggestion
          action: !args.options.userId && !args.options.userName && !args.options.groupId && !args.options.groupName ? 'selfActivate' : 'adminAssign',
```

What would help with readability is to destructure the options object first at the top of the commandAction function:

```ts
const [ userId, etc ] = args.options;
```